src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/register_app_interface_request.cc

```diff
@@ -1316,6 +1316,33 @@ void RegisterAppInterfaceRequest::SendSubscribeCustomButtonNotification() {
   CreateHMINotification(FunctionID::Buttons_OnButtonSubscription, msg_params);
 }
 
+bool RegisterAppInterfaceRequest::ApplicationWithSameNameAndID() {
```

Comment:
@ValeriiMalkov I see a function with a similar logic `CheckCoincidence` in this file. So do you really need this function?

Reply:
Function `CheckCoincidence` checks `appName` only. We need to check same  `appName`, `appID` and `connection_key`, because last parameter generates from `session_id` and `connection_id`. We need to check that isn't the same application sends `RegisterAppInterface` request, in this case `SDL` will send `APPLICATION_REGISTERED_ALREADY` response, in other case `DUPLICATE_NAME` .

Reply:
@AKalinich-Luxoft When the `SDL` received `appId` which was registered, it considers that's the application which was registered resend `RegisterAppInterface`, and `SDL` try to check that the application isn't in reconnection mode `IsAppInReconnectMode` method. If this method returns `false` we need to check that the same application and return `APPLICATION_REGISTERED_ALREADY` or another application via `connection_key`. If the `SDL` calls method `IsAppInReconnectMode` it means that it received the same `appID`. And it means i can to remove `appID` checking from `ApplicationWithSameNameAndID`, and check only `connection_key`. If it different `connection_key` i will call `CheckCoincidence` method for `appName` checking which will send `DUPLICATE_NAME`. 
I will do that, and push second commit.

Reply:
@AKalinich-Luxoft But i have other question. I think `SDL` should check is it same application or another before call method `IsAppInReconnectMode` via connection key, and if it another application  `SDL` should check `appName` only, in other case `SDL` checks reconnect mode`.

Reply:
@AKalinich-Luxoft Please see other implementation in second commit 14643632bb825b0987c9770f9fcc725cfebe2b21